src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -300,6 +311,65 @@ private IAIState findTrees()
         return LUMBERJACK_CHOP_TREE;
     }
 
+    @NotNull
+    @Override
+    protected IAIState dumpInventory() {
```

Comment:
I don't think all this dump inventory code here is necessary?

Reply:
The problem is that the worker can get stuck when a tree grows around him while he is choking down a tree, so when he then tries to bring the items back I need to check if he is stuck.
I can reduce it to only the first few lines and then call the super dumpInventory function

Reply:
did you test this? Because normally the AI should shut down completely when the head is in the leaves

Reply:
I tested it some minutes ago and its working fine :)

Reply:
shouldn't be necessary to copy all this code here though. If anything then you should put this in the walkToBlock override and not in copied functions.

Reply:
But then I need to copy everything to the EntityCitizenWalkToProxy class

Reply:
![image](https://user-images.githubusercontent.com/9266810/83512033-e11a3c80-a4cf-11ea-89b1-c35354b3337f.png)
The problem is that its final

Reply:
I will change that and override the function

Reply:
ye

Reply:
Still have the problem for the dumpInventory function, because its not calling the walkToBlock function of the ai
Instead its calling this function on the worker:
![image](https://user-images.githubusercontent.com/9266810/83512642-d0b69180-a4d0-11ea-8c1f-c6a73af0ae50.png)

Reply:
Am I allowed to change it in the abstract class to the walkToBlock funtion?

Reply:
and the walkToBlock on the AI, calls this also doesn't it? So we just redirect the dump call to this one.

Reply:
![image](https://user-images.githubusercontent.com/9266810/83512921-3a36a000-a4d1-11ea-83b1-664ca26c4b4b.png)
Its nearly the same

Reply:
it returns the opposite though. So you have to take that into account.

Reply:
I can just remove the ! in dumpInventory :)